src/checkbox-group.ts

```diff
@@ -380,7 +380,7 @@ export default class CheckboxGroup extends LitElement implements FormControl {
 
     const isValid = this.#internals.reportValidity();
 
-    // Ensures that getters referencing `this.validity.valid` are updated.
+    // Ensures that getters referencing `this.validity.valid` are re-run.
```

Comment:
It's not so much are the getters need to be updated as it is they need to be forced to run again.